federation/pkg/federation-controller/sync/controller.go

```diff
@@ -505,5 +516,23 @@ func clusterOperations(adapter federatedtypes.FederatedTypeAdapter, clusters []*
 			})
 		}
 	}
+
+	for _, cluster := range unselectedClusters {
```

Comment:
no change needed - it's just a tad annoying to need to iterate both the selected and unselected slices. Since it's faster than iterating through a map it's ok with me.